src/fdb5/api/LocalFDB.cc

```diff
@@ -46,14 +46,15 @@ using namespace eckit;
 
 
 namespace fdb5 {
+
 void LocalFDB::archive(const Key& key, const void* data, size_t length) {
 
     if (!archiver_) {
         LOG_DEBUG_LIB(LibFdb5) << *this << ": Constructing new archiver" << std::endl;
         archiver_.reset(new Archiver(config_));
     }
 
-    archiver_->archive(key, data, length);
+    archiver_->archive(key, data, length, callback_);
```

Comment:
It feels like the callback could be passed to the archiver at construction time. If necessary it could be updated? Or is there a reason to pass it in each time?

Reply:
I think you are right

Reply:
Moved to constructor